### PR TITLE
Remove Xor, chip away at Or (GEN-774)

### PR DESCRIPTION
**Pull request by sritchie**
This PR:

- removes `Xor` in favor of only using `Or` (my next PR will push `Or` down into `Choice`)
- modifies `from_mapping` to use `|=` instead of `^=`
- bans `|` between a `Choice` and a non-`Choice`. This should never occur EXCEPT in `Switch` and we handle that separately with a `SwitchChm`
- `|` between two choices pushes the `|` call down into the values
- throws a `TypeError` if a marginal is supplied to a `Target`; this is not supported!
- removes the `ravi_stack.ipynb` notebook from the docs site, as this was hitting that error

### CODE REVIEW

File: src/genjax/_src/core/generative/choice_map.py [hunk 1]
```diff
@@ -1454,11 +1457,8 @@ def build(v: T) -> ChoiceMap:
 
     def filter(self, selection: Selection) -> ChoiceMap:
         sel_check = selection.check()
-        masked = Mask.maybe_mask(self.v, sel_check)
-        if masked is None:
-            return ChoiceMap.empty()
-        else:
-            return ChoiceMap.choice(masked)
+        masked = Mask.build(self.v, sel_check)
```
Comment (sritchie) [inline src/genjax/_src/core/generative/choice_map.py:1460]: the simplification cases are already handled by `Choice.build`

File: src/genjax/_src/core/generative/choice_map.py [hunk 2]
```diff
@@ -1788,22 +1725,23 @@ def build(
...
-        match self.c1.get_value(), self.c2.get_value():
-            case None, b:
-                return b
-            case a, None:
-                return a
-            case a, b:
-                a = Mask.build(a)
-                b = Mask.build(b)
-                return (a | b).flatten()
+        return None
```
Comment (sritchie) [inline src/genjax/_src/core/generative/choice_map.py:1744]: it's no longer possible to build a value-producing `Or` instance.

File: src/genjax/_src/core/generative/choice_map.py [hunk 3]
```diff
@@ -1840,7 +1778,7 @@ def loop(inner: ChoiceMap, selection: Selection) -> Selection:
                 return acc
 
             case _:
-                return selection
+                raise ValueError(f"Unknown ChoiceMap type: {type(inner)}")
```
Comment (sritchie) [inline src/genjax/_src/core/generative/choice_map.py:1781]: I want to fail here, we should never hit this case

**Review by MathieuHuot — APPROVED** [review]
Looks good!

**Pull request merged**